consensus/src/main/java/org/apache/iotdb/consensus/ratis/Utils.java

```diff
@@ -265,4 +276,52 @@ public static void initRatisConfig(RaftProperties properties, RatisConfig config
     RaftServerConfigKeys.Rpc.setFirstElectionTimeoutMax(
         properties, config.getRpc().getFirstElectionTimeoutMax());
   }
+
+  /**
+   * Monitoring Ratis RaftLog total size. It will memorize the state of all files in the last update
+   * and calculates the diff incrementally each run.
+   */
+  static class RatisLogSizeMonitor {
+
+    private static final class DirectoryState {
+      private long size = 0;
+      private List<Path> memorizedFiles = Collections.emptyList();
+
+      private void update(long size, List<Path> latest) {
+        this.size = size;
+        this.memorizedFiles = latest;
+      }
+    }
+
+    private final ConcurrentHashMap<File, DirectoryState> directoryMap = new ConcurrentHashMap<>();
```

Comment:
Why need `ConcurrentHashMap` as there is only one thread?

Reply:
Changed to HashMap